README
TIMEZONE

Timezone aware date math and formats in JavaScript for Node.js and the browser.

    Read This:      http://bigeasy.github.com/timezone
    Source:         http://github.com/bigeasy/timezone
    Documentation:  http://github.com/bigeasy/timezone/wiki

    Build Status:   http://travis-ci.org/bigeasy/timezone
    Issues:         https://github.com/bigeasy/timezone/issues

    License:        The MIT License

PURPOSE

Timezone is a MicroJS library in pure JavaScript with no dependencies that
provides timezone aware date math and date formatting.

Timezone uses the IANA Database to determine the correct wall clock time
anywhere in the world for any time since the dawn of standardized time.

Timezone formats dates with a full implementation of `strftime` formats,
including the GNU `date` extensions.

Timezone represents time in POSIX time and local time using RFC 3999 date
strings.

Timezone is a full featured standards based time library in pure JavaScript for
under 3K minified and gzipped.

HAKCING

$ git clone --quiet git@github.com:bigeasy/timezone.git
$ cd timezone && make && npm install && npm test


CHANGELOG
- Version 0.0.16

 * Added `.js` suffix to unit tests.
 * Use `eggert/tz` GitHub repo as database source. #134.
 * Added a picture to the `README.md`. #121.

- Version 0.0.15 - Wed Jul 18 20:40:46 UTC 2012

 * Created a Markdown `README.md`. #118. #115.
 * Added many locales. #116.
 * Upgraded to Proof 0.0.15. #117. #113. #105. #102. #101.
 * Removed `main` property from `package.json`. #104.
 * Tidy whitespace and semi-colons. #97.
 * Plug all leaks of names into the global namespace. #107.
 * Run `utility/sizes` at Travis CI as part of continuous intgration build.
   #103.
 * Added `typeof` test against `module` in module loader preamble to
   `timezones.js` and `rfc822.js`. #94.
 * Merge `applicable` function into `find`. #95.
 * Merge `format` function into `convert`. #91.
 * Call `applicable` once. #88.
 * Merge `parse` function into `convert`. #87.
 * Merge `parseAdjustment` function into `convert`. #86.
 * Remove `slice` of arguments in `convert`. #93.
 * Delete `fields` variable from `actualize`. #92.
 * Merge `make` function into `convert`. #85.
 * Build on Node.js 0.8, add 0.8 to Travis CI. #90.
 * ixed `sysnopsis.js` assertions. #89.

- Version 0.0.14 - Mon Jun 25 21:02:24 UTC 2012

 * Publish tarball of data files at GitHub. #55.
 * Created AMD, JSONP and JSON files hosted at GitHub:Pages for use in web
   applications. #23.
 * Spell check `README`. #61.
 * Remove used data file from test directories. #67.
 * Make `zic` at Travis CI. #82.
 * Minify with lifted variables. #79.
 * Test minified code at Travis CI. #78.
 * Create `loaded.js` module to create a pre-loaded Timezone function. #42.
 * Simplify module loading boilerplate. #76.
 * Document date array and time zone format specifiers in `synopsis.js`. #50.
 * Remove `milli` synonym for `millisecond` math unit. #66.
 * Remove cached `Array.push` variable. #75.
 * Bump version string returned by Timezone function. #77. #72.
 * Reorganize `actualize` for byte savings. #63.
 * Revert to one based month in date array. #74.
 * Exclude non-JavaScript files, crufy files, from `slurp.js` indices. #71.
 * Remove over-zealous, bloaty argument verification. #70.

- Version 0.0.13 - Mon Jun 25 11:05:06 UTC 2012

 * Return internal state using function argument. #57.
 * Return library version when called with no arguments. #54.
 * Remove `posix` local variable from `convert` to save bytes. #62.
 * Add "@" to start of date array.
 * Add "z" to time zone data file array to differentiate zone entries from
   daylight savings time rules.

- Version 0.0.12 - Sun Jun 24 04:40:41 UTC 2012

 * Clean up commented out code. #65.
 * Retry unknown string arguments after all data sets are loaded. #59.
 * Remove `parseInt` to save bytes. #58.
 * Make `format` a non-member function to save bytes. #60.
 * Implement external RFC 822 parser function. #56.
 * Date arrays now fixed length. #52.
 * Replace ternary operator with short circuit or to save bytes. #53.
 * Reorder lexically similar code for better minification. #44.
 * Remove comparisons to zero to save bytes. #45.
 * UTC with partial didn't work if partial specified time zone name. #51.
 * Removed TODOs. #46.
 * Removed `require.date` property to save bytes. #47.
 * Accept numeric object for POSIX time. #48.
 * Build on OS X. #31.
 * Deleted defunct `index.js`. #49.
 * No longer parsing time only date strings. #30.
 * Implement `%^z`. #28.
 * Correctly format `%z` in UTC. #43.
 * Implement UK English, French and German locales. #26.
 * Convert Proof tests to JavaScript. #36.
 * Timezone zones and locales in package root. #34. #33.
 * Updates and revisions to `synopsis.js`.

- Version 0.0.11 - Thu Jun 14 18:04:39 UTC 2012

 * Correctly format `%_:z`. #40.
 * Correctly format `%-:z`. #39.
 * Tidy and sort out change log. #38.
 * Minor changes to save a few bytes when minified and zipped.
 * Accept `Date` as date argument.
 * Convert to plain text `README`. #37.
 * Combine `timezone.js` with `date.js`. #35. #25.

- Version 0.0.10 - Mon May 21 18:52:31 UTC 2012

 * Draft `synopsis.js`.
 * Remove CoffeeScript source. #27.
 * Correctly apply parsed timezone offset. #24.
 * Use specified padding GNU `strftime` extension. #22.
 * Implement linked zone files as copies. #11.
 * Replace binary search with skip-list search. #21.
 * Add heuristics assertions to pre-compiler. #16.
 * Use Ruby 1.9.3 to verify `%::z` formats using `tzdata` files.
 * Convert pre-compiler and library to CoffeeScript.
 * Add tests against `tzdata` to `npm test`. #17.
 * GNU `strftime` format parity. #9. #19.
 * Fixed `%C` which was off by machine zone offset.
 * Implemented parse RFC 3999 without year. #3. #7.
 * Added support for RequireJS AMD.
 * Changes to week of year functions for size savings. #20.
 * `tz` function is now the sole export of Timezone.
 * Accept date as an array of date fields.
 * Split less commonly used GNU `date` formats out into `date.js`.
 * Use `Object.create` to when creating partial `tz` functions.
 * Limit date parsing to RFC 3999.
 * Reject `Date` as date argument.

- Version 0.0.9 - Wed Apr 25 22:07:44 UTC 2012

 * Create `README.md` from `timezone.idl`. `.idl` is a document format for a
   document generator that I'm never going to write.
 * Move grumbling from `README.md` to `MOTIVATIONS.md`.
 * Add `milliseconds` as a synonym for `millis` to date math.
 * Configuration via partial functions.
 * Fix subtraction of years as per #10.
 * Attempt to create JSON as Olson by parsing the Olson files backwards in time.
 * Convert intermediate interval boundaries format to start time from end time.
 * Interpret slash delimited timezone abbreviations.
 * Interpret zone records with a specific savings offset.
 * Correctly convert between standard, wall-clock and POSIX time in preprocessing.
 * Correctly interpret rules that occur at the 24th hour; 24:00.
 * Correctly format `%Z`.
 * Use binary search against a generated table of rules.
 * Organize pre-compiler into a library and separate programs.

slurp.js
// Load all of the modules in a directory and any sub-directories. Used to
// implement `index.js` in the `locales` directory and the `zones` directory and
// sub-directories.
var fs = require("fs"), include, exclude;

function glob (exports, directory, include, exclude) {
  require("fs").readdirSync(directory).forEach(function (file) {
    var skip, path, stat;
    skip = skip || /^index.js|rfc822.js|slurp.js|synopsis.js|timezone.js|zones.js$/.test(file);
    skip = skip || /^[._]/.test(file);
    skip = skip || (exclude && exclude.test(file));
    skip = skip || include && ! include.test(file);
    if (! skip) {
      path = directory + "/" + file
      stat = fs.statSync(path);
      if (stat.isDirectory()) {
        glob(exports, path);
      } else {
        exports.push(require(path));
      }
    }
  });
}

if (/\/zones.js$/.test(__filename)) {
  include = /./;
  exclude = /^\w{2}_\w{2}.js$/;
} else if (/\/locales.js$/.test(__filename)) {
  include = /^\w{2}_\w{2}.js$/;
}

glob(module.exports = [], __dirname, include, exclude);


timezone.js
!function (definition) {
  if (typeof module == "object" && module.exports) module.exports = definition();
  else if (typeof define == "function" && typeof define.amd == "object") define(definition);
  else this.tz = definition();
} (function () {
  var __slice = [].slice, __push = [].push;
/*
  function die () {
    console.log.apply(console, __slice.call(arguments, 0));
    return process.exit(1);
  };

  function say () { return console.log.apply(console, __slice.call(arguments, 0)) }
*/
  function format (request, posix, rest) {
    var wallclock = new Date(convertToWallclock(request, posix));
    return rest.replace(/%([-0_^]?)(:{0,3})(\d*)(.)/g, function (matched, flag, colons, padding, specifier) {
      var f, value = matched, fill = "0";
      if (f = request[specifier]) {
        value = String(f.call(request, wallclock, posix, flag, (colons || "").length));
        if ((flag || f.style) == "_") fill = " ";
        pad = flag == "-" ? 0 : f.pad || 0;
        while (value.length < pad) value = fill + value;
        pad = flag == "-" ? 0 : padding || f.pad;
        while (value.length < pad) value = fill + value;
        if (specifier == "N" && pad < value.length) value = value.slice(0, pad);
        if (flag == "^") value = value.toUpperCase();
      }
      return value;
    });
  };

  function makeDate (request, date) {
    var posix = date[7], i;
    for (i = 0; i < 11; i++) date[i] = +(date[i] || 0); // conversion necessary for decrement
    --date[1];
    date = Date.UTC.apply(Date.UTC, date.slice(0, 8)) + -date[7] * (date[8] * 36e5 + date[9] * 6e4 + date[10] * 1e3);
    if (!posix) date = convertToPOSIX(request, date);
    if (date == null) throw new Error("invalid wall-clock time");
    return date;
  }

  function parse (pattern) {
    var date = [ "@" ], match;
    if (match = /^(\d{4})-(\d{2})-(\d{2})(?:[T\s](\d{2}):(\d{2})(?::(\d{2})(?:\.(\d+))?)?(Z|(([+-])(\d{2}(:\d{2}){0,2})))?)?$/.exec(pattern)) {
      __push.apply(date, match.slice(1, 8));
      if (match[9]) {
        __push.apply(date, [match[10] + 1]);
        __push.apply(date, match[11].split(/:/));
      } else if (match[8]) {
        __push.apply(date, [ 1 ]);
      }
      return date;
    }
  }

  function actualize (entry, rule, year) {
    var actualized, fields, date = rule.day[1];

    do {
      fields = new Date(Date.UTC(year, rule.month, Math.abs(date++)));
    } while (rule.day[0] < 7 && fields.getUTCDay() != rule.day[0])

    actualized = {
      clock: rule.clock,
      sortable: fields.getTime(),
      rule: rule,
      save: rule.save * 6e4,
      offset: entry.offset
    };

    actualized[actualized.clock] = fields.getTime() + rule.time * 6e4;

    if (actualized.clock == "posix") {
      actualized.wallclock = actualized.posix + (entry.offset + rule.saved);
    } else {
      actualized.posix = actualized.wallclock - (entry.offset + rule.saved);
    }

    return actualized;
  }

  function applicable (entry, rules, actualized, time) {
    var last, i, I, j, year = new Date(time).getUTCFullYear();
    for (j = year + 1; j >= year - 1; --j) {
      for (i = 0, I = rules.length; i < I; i++) {
          if (rules[i].from <= j && j <= rules[i].to) {
            actualized.push(actualize(entry, rules[i], j));
          } else if (rules[i].to < j) {
            last = rules[i].to;
            break;
        }
      }
    }
    return last;
  }

  function find (request, clock, time) {
    var i, I, entry, year = new Date(time).getUTCFullYear(), found, zone = request[request.zone], actualized = [], to, abbrevs;
    for (i = 1, I = zone.length; i < I; i++) if (zone[i][clock] <= time) break;
    entry = zone[i];
    if (entry.rules) {
      rules = request[entry.rules];
      to = applicable(entry, rules, actualized, time);
      if (to != null) applicable(entry, rules, actualized, Date.UTC(to, 5));
      actualized.sort(function (a, b) { return a.sortable - b.sortable });
      for (i = 0, I = actualized.length; i < I; i++) {
        if (time >= actualized[i][clock] && actualized[i][actualized[i].clock] > entry[actualized[i].clock]) found = actualized[i];
      }
    }
    if (found) {
      if (abbrevs = /^(.*)\/(.*)$/.exec(entry.format)) {
        found.abbrev = abbrevs[found.save ? 2 : 1];
      } else {
        found.abbrev = entry.format.replace(/%s/, found.rule.letter);
      }
    }
    return found || entry;
  }

  function convertToWallclock (request, posix) {
    if (request.zone == "UTC") return posix;
    request.entry = find(request, "posix", posix);
    return posix + request.entry.offset + request.entry.save;
  };

  function convertToPOSIX (request, wallclock) {
    if (request.zone == "UTC") return wallclock;

    var entry, diff;
    request.entry = entry = find(request, "wallclock", wallclock);
    diff = wallclock - entry.wallclock;

    return 0 < diff && diff < entry.save ? null : wallclock - entry.offset - entry.save;
  };

  function parseAdjustment (pattern) {
    var match = UNIT_RE.exec(pattern);
    return match && function (request, posix) { return adjust(request, posix, match) }
  };

  function adjust (request, posix, match) {
    var increment = +(match[1] + 1) // conversion necessary for week day addition
      , offset = match[2] * increment
      , index = UNITS.indexOf(match[3].toLowerCase())
      , date
      ;
    if (index > 9) {
      posix += offset * TIME[index - 10];
    } else {
      date = new Date(convertToWallclock(request, posix));
      if (index < 7) {
        while (offset) {
          date.setUTCDate(date.getUTCDate() + increment);
          if (date.getUTCDay() == index) offset -= increment;
        }
      } else if (index == 7) {
        date.setUTCFullYear(date.getUTCFullYear() + offset);
      } else if (index == 8) {
        date.setUTCMonth(date.getUTCMonth() + offset);
      } else {
        date.setUTCDate(date.getUTCDate() + offset);
      }
      if ((posix = convertToPOSIX(request, date.getTime())) == null) {
        posix = convertToPOSIX(request, date.getTime() + 864e5 * increment) - 864e5 * increment;
      }
    }
    return posix;
  };

  function convert (vargs) {
    if (!vargs.length) return "0.0.11";

    var i, I, argument, date, type
      , request = Object.create(this)
      , zone, locale
      , adjustments = []
      , parsed
      , extra = []
      ;

    for (i = 0; vargs.length; i++) { // leave the for loop alone, it works.
      argument = vargs.shift();
      // https://twitter.com/bigeasy/status/215112186572439552
      if (argument == null) {
        throw new Error("null argument");
      } else if (Array.isArray(argument)) {
        if (!i && argument[0] == "@") {
          date = argument;
        } else {
          vargs.unshift.apply(vargs, argument);
          i--;
        }
      } else if (isNaN(argument) && (i || argument != "*")) {
        type = typeof argument;
        if (type == "string") {
          if (~argument.indexOf("%")) {
            request.format = argument;
          } else if (!i && (parsed = parse(argument))) {
            date = parsed;
          } else if (/^\w{2}_\w{2}$/.test(argument)) {
            if (locale) extra.push(argument);
            else locale = argument;
          } else if (parsed = parseAdjustment(argument)) {
            adjustments.push(parsed);
          } else if (/^[-A-Za-z\/_]+$|^\w{3}\d\w{3}$/.test(argument)) {
            if (!zone) zone = argument;
            else extra.push(argument);
          } else {
            throw new Error("invalid argument: " + argument);
          }
        } else if (type == "function") {
          if (parsed = argument.call(request)) return parsed;
        } else if (/^\w{2}_\w{2}$/.test(argument.name)) {
          request[argument.name] = argument;
        } else if (argument.zones) {
          for (var key in argument.zones) request[key] = argument.zones[key];
          for (var key in argument.rules) request[key] = argument.rules[key];
        }
      } else if (!i) {
        date = argument;
      } else {
        throw new Error("invalid argument: " + argument);
      }
    }

    if (locale) {
      if (!request[locale]) throw new Error("invalid locale: " + locale);
      request.locale = locale;
    }

    if (zone) {
      if (!request[zone] || request[zone][0] != "z") throw new Error("invalid zone: " + zone);
      request.zone = zone;
    }

    if (extra.length) return request.convert(date == null ? extra : [ date, extra ]);

    if (date != null) {
      if (date == "*") {
        date = request.clock();
      } else if (Array.isArray(date)) {
        if (isNaN(date = makeDate(request, date.slice(1))))
          throw new Error("invalid date");
      } else {
        date = Math.floor(date);
      }

      for (i = 0, I = adjustments.length; i < I; i++) {
        date = adjustments[i](request, date);
      }

      return request.format ? format(request, date, request.format) : date;
    }

    return function () { return request.convert(__slice.call(arguments, 0)) };
  };

  var context =
    { zone: "UTC"
    , entry: { abbrev: "UTC", offset: 0, save: 0 }
    , UTC: [ "z" ]
    , clock: function () { return +(new Date()) }
    , convert: convert
    , z: function(date, posix, flag, delimiters) {
        var offset = this.entry.offset + this.entry.save
          , seconds = Math.abs(offset / 1000), parts = [], part = 3600, i, z;
        for (i = 0; i < 3; i++) {
          parts.push(("0" + Math.floor(seconds / part)).slice(-2));
          seconds %= part;
          part /= 60;
        }
        if (flag == "^" && !offset) return "Z";
        if (flag == "^") delimiters = 3;
        if (delimiters == 3) {
          z = parts.join(":");
          z = z.replace(/:00$/, "");
          if (flag != "^") z = z.replace(/:00$/, "");
        } else if (delimiters) {
          z = parts.slice(0, delimiters + 1).join(":");
          if (flag == "^") z = z.replace(/:00$/, "");
        } else {
          z = parts.slice(0, 2).join("");
        }
        z = (offset < 0 ? "-" : "+") + z;
        z = z.replace(/([-+])(0)/, { "_": " $1", "-": "$1" }[flag] || "$1$2");
        return z;
      }
    , "%": function(date) { return "%" }
    , n: function (date) { return "\n" }
    , t: function (date) { return "\t" }
    , U: function (date) { return weekOfYear(date, 0) }
    , W: function (date) { return weekOfYear(date, 1) }
    , V: function (date) { return isoWeek(date)[0] }
    , G: function (date) { return isoWeek(date)[1] }
    , g: function (date) { return isoWeek(date)[1] % 100 }
    , j: function (date) { return Math.floor((date.getTime() - Date.UTC(date.getUTCFullYear(), 0)) / 864e5) + 1 }
    , s: function (date) { return Math.floor(date.getTime() / 1000) }
    , C: function (date) { return Math.floor(date.getUTCFullYear() / 100) }
    , N: function (date) { return date.getTime() % 1000 * 1000000 }
    , m: function (date) { return date.getUTCMonth() + 1 }
    , Y: function (date) { return date.getUTCFullYear() }
    , y: function (date) { return date.getUTCFullYear() % 100 }
    , H: function (date) { return date.getUTCHours() }
    , M: function (date) { return date.getUTCMinutes() }
    , S: function (date) { return date.getUTCSeconds() }
    , e: function (date) { return date.getUTCDate() }
    , d: function (date) { return date.getUTCDate() }
    , u: function (date) { return date.getUTCDay() || 7 }
    , w: function (date) { return date.getUTCDay() }
    , l: function (date) { return date.getUTCHours() % 12 || 12 }
    , I: function (date) { return date.getUTCHours() % 12 || 12 }
    , k: function (date) { return date.getUTCHours() }
    , Z: function (date) { return this.entry.abbrev }
    , a: function (date) { return this[this.locale].day.abbrev[date.getUTCDay()] }
    , A: function (date) { return this[this.locale].day.full[date.getUTCDay()] }
    , h: function (date) { return this[this.locale].month.abbrev[date.getUTCMonth()] }
    , b: function (date) { return this[this.locale].month.abbrev[date.getUTCMonth()] }
    , B: function (date) { return this[this.locale].month.full[date.getUTCMonth()] }
    , P: function (date) { return this[this.locale].meridiem[Math.floor(date.getUTCHours() / 12)].toLowerCase() }
    , p: function (date) { return this[this.locale].meridiem[Math.floor(date.getUTCHours() / 12)] }
    , R: function (date, posix) { return this.convert([ posix, "%H:%M" ]) }
    , T: function (date, posix) { return this.convert([ posix, "%H:%M:%S" ]) }
    , D: function (date, posix) { return this.convert([ posix, "%m/%d/%y" ]) }
    , F: function (date, posix) { return this.convert([ posix, "%Y-%m-%d" ]) }
    , x: function (date, posix) { return this.convert([ posix, this[this.locale].date ]) }
    , r: function (date, posix) { return this.convert([ posix, this[this.locale].time12 ]) }
    , X: function (date, posix) { return this.convert([ posix, this[this.locale].time24 ]) }
    , c: function (date, posix) { return this.convert([ posix, this[this.locale].dateTime ]) }
    , locale: "en_US"
    , en_US: {
        date: "%m/%d/%Y",
        time24: "%I:%M:%S %p",
        time12: "%I:%M:%S %p",
        dateTime: "%a %d %b %Y %I:%M:%S %p %Z",
        meridiem: [ "AM", "PM" ],
        month: {
          abbrev: "Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec".split("|"),
          full: "January|February|March|April|May|June|July|August|September|October|November|December".split("|")
        },
        day: {
          abbrev: "Sun|Mon|Tue|Wed|Thu|Fri|Sat".split("|"),
          full: "Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday".split("|")
        }
      }
    };
  var UNITS = "Sunday|Monday|Tuesday|Wednesday|Thursday|Friday|Saturday|year|month|day|hour|minute|second|milli|millisecond"
    , UNIT_RE = new RegExp("^\\s*([+-])(\\d+)\\s+(" + UNITS + ")s?\\s*$", "i")
    , TIME = [ 36e5, 6e4, 1e3, 1, 1 ]
    ;
  UNITS = UNITS.toLowerCase().split("|");

  "dmHMSUWVgCIky".replace(/./g, function (e) { context[e].pad = 2 });

  context.N.pad = 9;
  context.j.pad = 3;

  context.k.style = "_";
  context.l.style = "_";
  context.e.style = "_";

  function weekOfYear (date, startOfWeek) {
    var diff, nyd, weekStart;
    nyd = new Date(Date.UTC(date.getUTCFullYear(), 0));
    diff = Math.floor((date.getTime() - nyd.getTime()) / 864e5);
    if (nyd.getUTCDay() == startOfWeek) {
      weekStart = 0;
    } else {
      weekStart = 7 - nyd.getUTCDay() + startOfWeek;
      if (weekStart == 8) {
        weekStart = 1;
      }
    }
    return diff >= weekStart ? Math.floor((diff - weekStart) / 7) + 1 : 0;
  }

  function isoWeek (date) {
    var nyd, nyy, week;
    nyy = date.getUTCFullYear();
    nyd = new Date(Date.UTC(nyy, 0)).getUTCDay();
    week = weekOfYear(date, 1) + (nyd > 1 && nyd <= 4 ? 1 : 0);
    if (!week) {
      nyy = date.getUTCFullYear() - 1;
      nyd = new Date(Date.UTC(nyy, 0)).getUTCDay();
      week = nyd == 4 || (nyd == 3 && new Date(nyy, 1, 29).getDate() == 29) ? 53 : 52;
      return [week, date.getUTCFullYear() - 1];
    } else if (week == 53 && !(nyd == 4 || (nyd == 3 && new Date(nyy, 1, 29).getDate() == 29))) {
      return [1, date.getUTCFullYear() + 1];
    } else {
      return [week, date.getUTCFullYear()];
    }
  }

  return function () { return context.convert(__slice.call(arguments, 0)) };
});


locales/es_ES.js
// dom lun mar mié jue vie sáb
// domingo lunes martes miércoles jueves viernes sábado
// ene feb mar abr may jun jul ago sep oct nov dic
// enero febrero marzo abril mayo junio julio agosto septiembre octubre noviembre diciembre
// 
// 
// 
// 
// 03/09/00
// 03/09/00
// 08:05:04
// 23:05:04
// dom 03 sep 2000 08:05:04 UTC
// dom 03 sep 2000 23:05:04 UTC
// 
module.exports = {
  "name": "es_ES",
  "day": {
    "abbrev": [
      "dom",
      "lun",
      "mar",
      "mié",
      "jue",
      "vie",
      "sáb"
    ],
    "full": [
      "domingo",
      "lunes",
      "martes",
      "miércoles",
      "jueves",
      "viernes",
      "sábado"
    ]
  },
  "month": {
    "abbrev": [
      "ene",
      "feb",
      "mar",
      "abr",
      "may",
      "jun",
      "jul",
      "ago",
      "sep",
      "oct",
      "nov",
      "dic"
    ],
    "full": [
      "enero",
      "febrero",
      "marzo",
      "abril",
      "mayo",
      "junio",
      "julio",
      "agosto",
      "septiembre",
      "octubre",
      "noviembre",
      "diciembre"
    ]
  },
  "meridiem": [
    {},
    {}
  ],
  "date": "%d/%m/%y",
  "time24": "%H:%M:%S",
  "dateTime": "%a %d %b %Y %H:%M:%S %Z"
};
